### ISSUE TITLE
Add Deno File System Support

### ISSUE THREAD

**Issue by okikio** [unjs/unstorage#230]
### Describe the feature

Deno has a built-in file system storage API's, it would be great if we can use it with unstorage!

### Additional information

- [ ] Would you be willing to help implement this feature?

**Comment by okikio** [comment]
I would love to help implementing this feature but I'm unfortunately unavailble

**Comment by so1ve** [comment]
I think you can simply import unstorage via `npm:unstorage` and the file system will be polyfilled

**Comment by okikio** [comment]
The goal is to support the Deno API is places other than Deno, e.g. there's a Deno library using the Deno FS API, this would allow said library to be supported in the browser or on Node

**Comment by pi0** [comment]
Deno v2 has a decent support for Node.js compatibility now so it immediately doesn't makes much sense to put effort. We recently also added support for Deno KV (both Node and Deno).

In the future, fs driver might be rebased to use (upcoming) unfs to directly use native runtime APIs.

**Comment by okikio** [comment]
Is unfs going to be a separate project?

**Comment by pi0** [comment]
yes (and to be used for fs driver internally when ready) but mostly for semantic reasons rather than functionality (as Deno supports Node.js fs API today)

**Comment by okikio** [comment]
The Node.js fs API is an abstraction over Deno native implementation with perf. impacts, it's actually the same for Bun, Bun slows down quite a bit when using Node.js's fs APIs. Supporting the native runtime fs APIs which are built on web stds. anyway is still necessary imo.

**Comment by pi0** [comment]
Deno/Bun fs APIs are not based on Web Standards (there is no webstandard).

There might be indeed slight performance improvement in this which is one of the reasons planning to work on unfs migration, however please notice it is not remarkable perf gain.

**Comment by okikio** [comment]
Oh really? Last I tested the perf., the difference was fairly notable 🤔

**Comment by pi0** [comment]
Please do share any benchmarks if you found there is significant difference between Deno FS api and Deno v2 node compat (you can always easily create a custom unstorage driver)

**Comment by okikio** [comment]
I'll benchmark the APIs again, if I notice a large difference I'll keep you updated

**Comment by pi0** [comment]
Made a dummy [bench](https://gist.github.com/pi0/019ab8ad75ce2b2ef79adeb60107cead) to compare with node-fs. 

```
clk: ~4.39 GHz
cpu: Apple M4 Pro
runtime: deno 2.1.4 (aarch64-apple-darwin)

benchmark                   avg (min … max) p75   p99    (min … top 1%)
------------------------------------------- -------------------------------
node-api                      72.49 µs/iter  74.29 µs    ▅█▂               
                     (58.08 µs … 335.67 µs) 111.46 µs ▂▃▂███▇▅▃▃▂▂▂▁▁▁▁▁▁▁▁
deno-api                      81.30 µs/iter  82.29 µs    █▃                
                     (63.67 µs … 574.21 µs) 144.21 µs ▁▁███▆▃▃▂▂▁▁▁▁▁▁▁▁▁▁▁

summary
  node-api
   1.12x faster than deno-api
```

(will double check later I might have been missing a deno API specially for `existsSync` replacement however overall Node API on deno somehow is even faster today!)
